settings_test.go

```diff
@@ -0,0 +1,269 @@
+package solus
+
+import (
+	"context"
+	"encoding/json"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"testing"
+)
+
+func TestSettings_Get(t *testing.T) {
+	s := startTestServer(t, func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/settings", r.URL.Path)
+		assert.Equal(t, http.MethodGet, r.Method)
+
+		writeResponse(t, w, http.StatusCreated, fakeSettingsRawJSON)
+	})
+	defer s.Close()
+
+	actual, err := createTestClient(t, s.URL).Settings.Get(context.Background())
+	require.NoError(t, err)
+
+	fakeSettingsResponse := struct {
+		Data Settings `json:"data"`
+	}{}
+	require.NoError(t, json.Unmarshal(fakeSettingsRawJSON, &fakeSettingsResponse))
+
+	require.Equal(t, fakeSettingsResponse.Data, actual)
+}
+
+func TestSettings_ChangeHostname(t *testing.T) {
+	host := "new-hostname.tld"
+	data := struct {
+		Hostname string `json:"hostname"`
+	}{
+		Hostname: host,
+	}
+
+	s := startTestServer(t, func(w http.ResponseWriter, r *http.Request) {
+		assert.Equal(t, "/settings", r.URL.Path)
+		assert.Equal(t, http.MethodPatch, r.Method)
+		assertRequestBody(t, r, data)
+
+		writeResponse(t, w, http.StatusCreated, fakeSettingsRawJSON)
+	})
+	defer s.Close()
+
+	request := SettingsUpdateRequest{
+		Hostname: &host,
+	}
+	actual, err := createTestClient(t, s.URL).Settings.Patch(context.Background(), request)
+	require.NoError(t, err)
+
+	fakeSettingsResponse := struct {
+		Data Settings `json:"data"`
+	}{}
+	require.NoError(t, json.Unmarshal(fakeSettingsRawJSON, &fakeSettingsResponse))
+
+	require.Equal(t, fakeSettingsResponse.Data, actual)
+}
+
+var fakeSettingsRawJSON = []byte(`{
```

Comment:
Why we should use a raw JSON instead of out own structure?

Reply:
Raw response from real instance helps to find type mismatch errors. Own structure can't be used for testing JSON unmarshaling.